src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbVacancyReviewRepository.cs

```diff
@@ -22,6 +22,37 @@ public MongoDbVacancyReviewRepository(ILogger<MongoDbVacancyReviewRepository> lo
         {
         }
 
+        public async Task<List<VacancyReviewSearch>> SearchAsync(long vacancyReference)
+        {
+            var filterBuilder = Builders<VacancyReview>.Filter;
+
+            var filter = (filterBuilder.Eq(r => r.Status, ReviewStatus.PendingReview)
+                         | filterBuilder.Eq(r => r.Status, ReviewStatus.UnderReview)) 
+                         & filterBuilder.Eq(r => r.VacancyReference, vacancyReference);
+            var collection = GetCollection<VacancyReview>();
+            
+            var result = await RetryPolicy
+                .ExecuteAsync( 
+                    context => collection
+                        .Find(filter)
+                        .Project(r => new VacancyReviewSearch()
```

Comment:
This ,ethod should return the VacancyReview and then that will get mapped to the VacancyReviewSearch from the consuming code.

Reply:
With VacancySnapshot added, Vacancy Review is now a bulky document, hence I am projecting to retrieve only required fields.